lib/Doctrine/ORM/Repository/InvalidFindByInverseAssociation.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Repository;
+
+use Doctrine\ORM\RepositoryException;
+
+final class InvalidFindByInverseAssociation extends \BadMethodCallException implements RepositoryException
+{
+    public static function becauseIsInverseAssociation(
```

Comment:
@Ocramius @SenseException can you please agree with a "because" vs "from" naming recommendation before I migrate all the exceptions?

Reply:
Like I previously wrote, "because" sounds too funny, so I can agree on a different prefix. If we go with my exception name suggestions of a previous comment about this class name, we can create a method like `InvalidAssociationSide::fromInverseSide()` or `InvalidAssociation::fromInverseSideUsage()`. We can recombine method name and class name between those two method-suggestions.

Reply:
I don't think there is a point to using named constructor if we don't have several of them. I'll therefore go with your second proposition.

Reply:
Or rather, `InvalidFindByCall::fromInverseSideUsage`

Reply:
I think I'm gonna cram `findByRequiresParameter` in there, Ocramius said those should be almost all separate exceptions, I think using the "almost" card here is right.

Reply:
`fromMissingParameter` sounds weird though, maybe `because` is not that bad 🤔

Reply:
How about `onMissingParameter`?

Reply:
> I think I'm gonna cram findByRequiresParameter in there, Ocramius said those should be almost all separate exceptions, I think using the "almost" card here is right.

as @ostrolucky pointed out, this one is not about `findBy`, but about a magic call, so I guess I'm gonna put it in `InvalidMagicCall` instead

Reply:
`EvilMagicCall`  😆
It sounds right that it belongs into an exception for magic calls.
About just one needed named constructor: They become or already are a standard in Doctrine, so it would be weird if we have one case where an instance was created with `new`. Especially if that exception class might need a second exception case one day, we have to look that the one with `new` will get an own named constructor too.